mobile/src/main/java/rs/readahead/washington/mobile/mvp/presenter/LocationGettingPresenter.java

```diff
@@ -179,6 +192,21 @@ private static class MyLocationCallback extends LocationCallback {
         public void onLocationResult(LocationResult locationResult) {
             Location location = locationResult.getLastLocation();
 
+            if (presenter.get() != null) {
+                presenter.get().sendLocation(location);
+            }
+        }
+    }*/
+
+    private static class MyLocationConsumer implements IMyLocationConsumer {
+        private WeakReference<LocationGettingPresenter> presenter;
+
+        MyLocationConsumer(LocationGettingPresenter presenter) {
+            this.presenter = new WeakReference<>(presenter);
+        }
+
+        @Override
```

Comment:
what is the idea here ? should we create this as a separate class and initialize in the view?

Reply:
I wanted to leave the listener as it was before for compatibility reasons, but there's no need for it...